sdk/digitaltwins/digitaltwins/review/digitaltwins.api.md

```diff
@@ -0,0 +1,354 @@
+
+// @public
+export interface DigitalTwinsUpdateHeaders {
+    etag?: string;
+}
+
+// @public
+export interface DigitalTwinsUpdateOptionalParams extends coreHttp.OperationOptions {
+    ifMatch?: string;
+}
+
+// @public
+export interface DigitalTwinsUpdateRelationshipHeaders {
+    etag?: string;
+}
+
+// @public
+export interface DigitalTwinsUpdateRelationshipOptionalParams extends coreHttp.OperationOptions {
+    ifMatch?: string;
+    patchDocument?: any[];
+}
+
+// @public
+export type DigitalTwinsUpdateRelationshipResponse = DigitalTwinsUpdateRelationshipHeaders & {
+    _response: coreHttp.HttpResponse & {
+        parsedHeaders: DigitalTwinsUpdateRelationshipHeaders;
+    };
+};
+
+// @public
+export type DigitalTwinsUpdateResponse = DigitalTwinsUpdateHeaders & {
+    _response: coreHttp.HttpResponse & {
+        parsedHeaders: DigitalTwinsUpdateHeaders;
+    };
+};
+
+// @public
+export interface EventRoute {
+    endpointName: string;
+    filter?: string;
+    readonly id?: string;
+}
+
+// @public
+export interface EventRouteCollection {
+    nextLink?: string;
+    value?: EventRoute[];
+}
+
+// @public
+export interface EventRoutesAddOptionalParams extends coreHttp.OperationOptions {
+    eventRoute?: EventRoute;
+}
+
+// @public
+export type EventRoutesGetByIdResponse = EventRoute & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: EventRoute;
+    };
+};
+
+// @public
+export type EventRoutesListNextResponse = EventRouteCollection & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: EventRouteCollection;
+    };
+};
+
+// @public
+export interface EventRoutesListOptionalParams extends coreHttp.OperationOptions {
+    eventRoutesListOptions?: EventRoutesListOptions;
+}
+
+// @public
+export interface EventRoutesListOptions {
+    maxItemCount?: number;
+}
+
+// @public
+export interface IncomingRelationship {
+    relationshipId?: string;
+    relationshipLink?: string;
+    relationshipName?: string;
+    sourceId?: string;
+}
+
+// @public
+export interface IncomingRelationshipCollection {
+    nextLink?: string;
+    // (undocumented)
+    value?: IncomingRelationship[];
+}
+
+// @public
+export interface ModelData {
+    decommissioned?: boolean;
+    description?: {
+        [propertyName: string]: string;
+    };
+    displayName?: {
+        [propertyName: string]: string;
+    };
+    id: string;
+    model?: any;
+    uploadTime?: Date;
+}
+
+// @public
+export interface PagedModelDataCollection {
+    nextLink?: string;
+    value?: ModelData[];
+}
+
+// @public
+export interface QueryQueryTwinsHeaders {
+    queryCharge?: number;
+}
+
+// @public
+export type QueryQueryTwinsResponse = QueryQueryTwinsHeaders & QueryResult & {
+    _response: coreHttp.HttpResponse & {
+        bodyAsText: string;
+        parsedBody: QueryResult;
+        parsedHeaders: QueryQueryTwinsHeaders;
+    };
+};
+
+// @public
+export interface QueryResult {
```

Comment:
should this be a generic so the consumer can type `items`? Same thought for `RelationshipCollection` below

Reply:
I don't understand what you mean. What about the continuationToken?

---
In reply to: [470198848](https://github.com/Azure/azure-sdk-for-js/pull/10560#discussion_r470198848) [](ancestors = 470198848)

Reply:
what about something like

```ts

export interface QueryResult<T> {
    continuationToken?: string;
    items?: T[];
}
```